tests/test_macos_backend.py

```diff
@@ -43,3 +50,18 @@ def test_osx_token_cache_roundtrip():
         assert token1['access_token'] == token2['access_token']
     finally:
         shutil.rmtree(test_folder, ignore_errors=True)
+
+@pytest.mark.skipif(
+    is_running_on_travis_ci, reason="Requires manual testing")
+def test_macos_no_keychain_entry_exists_before_first_use():
+    test_folder = tempfile.mkdtemp(prefix="msal_extension_test_windows_token_cache_roundtrip")
+    cache_file = os.path.join(test_folder, 'msal.cache')
+    open(cache_file, 'w+')
+    try:
+        # Make sure key chain entry does not already exist for below service name
+        persistence = KeychainPersistence(cache_file, "my_service_name", "my_account_name")
+        app = msal.PublicClientApplication(
```

Comment:
Let's move these test cases into test_persistence.py

Also, is it possible to have these 2 new test cases NOT depend on msal.PublicClientApplication?